### ISSUE TITLE
Error bundling cross-compiled application which is unsupported for current CPU architecture

### ISSUE THREAD

**Issue by ysmoradi** [velopack/velopack#10]
I've published the app using:
```bash
dotnet publish TodoSample.Client.Windows.csproj -c Release -o ./publish -r win-arm64
```
Later I run:
```bash
dotnet vpk pack -u TodoSample.Client.Windows -v 1.0.0 -p .\publish -e TodoSample.Client.Windows.exe -r win-x86 --framework net8.0.1-x86-desktop,webview2 --icon .\wwwroot\favicon.ico --packTitle TodoSample
```
Then I receive the following error:
```bash
[19:36:38 FTL] Command ExecutePackWindows had an exception.
    Win32Exception: An error occurred trying to start process
    'C:\Workspace\temp\TodoSample\src\Client\TodoSample.Client.Windows\publish\TodoSample.Client.Windows.exe' with
    working directory 'C:\Workspace\temp\TodoSample\src\Client\TodoSample.Client.Windows'. The specified executable is
    not a valid application for this OS platform.
      at bool StartWithCreateProcess(ProcessStartInfo startInfo)
      at Process Start(ProcessStartInfo startInfo)
      at string Output(ProcessStartInfo psi, int timeoutMs) in ProcessExtensions.cs:67
      at async Task Run(T options) in PackageBuilder.cs:83
      at async Task<int> CreateAndExecuteAsync<T>(string commandName, T options) in RunnerFactory.cs:26
```
It seems velopack tries to start the app, which is not possible, because arm64 won't run in x64 machine.
But what's the benefit of running the app?
The same issue is making my CD pipeline more complex, because I need to install some stuffs to make sure x86 version of the app runs on my agent which is X64.

**Comment by caesay** [comment]
I'll have to think about this. Currently, the builder runs the executable to receive some metadata from the VelopackApp, and verify that the hooks are set up correctly.

**Comment by ysmoradi** [comment]
It's really a blocker, no CD agent is able to run arm64 😫
So please leave verification to the developer and let us provide any metadata you need with velopack cli

**Comment by caesay** [comment]
I don't want to remove this check entirely, because it can help to catch mistakes that developers make when setting up the VelopackApp.Build().Run() code at the beginning of their app. These mistakes happen a lot, and result in really weird behavior on Windows, and ends up being support requests and such here. However, I have disabled the check when the main binary is detected to be a different CPU architecture than the current system architecutre is. This will be released shortly, I'm just waiting for the CI build to complete.

**Comment by ysmoradi** [comment]
I was fixed at 0.0.219, but now it's broken at 0.0.226

**Comment by caesay** [comment]
I have removed this check entirely for now, and I will implement this check again differently in a future version.
